[Bug]: Unable to get Artifacts UI working locally

### What happened?

I am running LibreChat v0.7.6 locally using Docker by pulling the specific image from the repository. I have LibreChat use Bedrock models on my own AWS account and using the latest Sonnet model in this case. (I also tried the same with Anthropic API, but getting same errors locally)

When i try a simple prompt to initiate Artifacts UI, i get this error which is stopping the Artifacts UI to work properly.

```
TypeError: Cannot read properties of undefined (reading 'size')
    at Dr (http://localhost:3080/assets/index.BmzNNmTG.js:3152:82907)
    at I7 (http://localhost:3080/assets/vendor.9TVOqDrM.js:38:19558)
    at qee (http://localhost:3080/assets/vendor.9TVOqDrM.js:40:44172)
    at Uee (http://localhost:3080/assets/vendor.9TVOqDrM.js:40:39877)
    at mSe (http://localhost:3080/assets/vendor.9TVOqDrM.js:40:39800)
    at v2 (http://localhost:3080/assets/vendor.9TVOqDrM.js:40:39649)
    at j$ (http://localhost:3080/assets/vendor.9TVOqDrM.js:40:36005)
    at Fee (http://localhost:3080/assets/vendor.9TVOqDrM.js:40:34951)
    at N (http://localhost:3080/assets/vendor.9TVOqDrM.js:25:1535)
    at MessagePort.u (http://localhost:3080/assets/vendor.9TVOqDrM.js:25:1894)
```

Please check this video,

https://github.com/user-attachments/assets/1215b948-e9ce-4d28-aa27-c50e227db444





### Steps to Reproduce

1. Pull v0.7.6 images - `docker pull ghcr.io/danny-avila/librechat:v0.7.6`
2. Start containers - `docker compose up -d`
3. Select a model and ask this prompt - "Create a playable snake game"

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
vendor.9TVOqDrM.js:40 TypeError: Cannot read properties of undefined (reading 'size')
    at Dr (index.BmzNNmTG.js:3152:82907)
    at I7 (vendor.9TVOqDrM.js:38:19558)
    at qee (vendor.9TVOqDrM.js:40:44172)
    at Uee (vendor.9TVOqDrM.js:40:39877)
    at mSe (vendor.9TVOqDrM.js:40:39800)
    at v2 (vendor.9TVOqDrM.js:40:39649)
    at j$ (vendor.9TVOqDrM.js:40:36005)
    at Fee (vendor.9TVOqDrM.js:40:34951)
    at N (vendor.9TVOqDrM.js:25:1535)
    at MessagePort.u (vendor.9TVOqDrM.js:25:1894)
M$ @ vendor.9TVOqDrM.js:40Understand this errorAI
vendor.9TVOqDrM.js:66 React Router caught the following error during render TypeError: Cannot read properties of undefined (reading 'size')
    at Dr (index.BmzNNmTG.js:3152:82907)
    at I7 (vendor.9TVOqDrM.js:38:19558)
    at qee (vendor.9TVOqDrM.js:40:44172)
    at Uee (vendor.9TVOqDrM.js:40:39877)
    at mSe (vendor.9TVOqDrM.js:40:39800)
    at v2 (vendor.9TVOqDrM.js:40:39649)
    at j$ (vendor.9TVOqDrM.js:40:36005)
    at Fee (vendor.9TVOqDrM.js:40:34951)
    at N (vendor.9TVOqDrM.js:25:1535)
    at MessagePort.u (vendor.9TVOqDrM.js:25:1894) Object
componentDidCatch @ vendor.9TVOqDrM.js:66Understand this errorAI
babel-transpiler.2641e80b.worker.js:1 TypeError: Cannot assign to read only property 'message' of object 'SyntaxError: /App.tsx: Unexpected token (2:0)

  1 | import { useState,
> 2 |
    | ^'
    at babel-transpiler.2641e80b.worker.js:1:1089686
    at Generator.next (<anonymous>)
    at n (babel-transpiler.2641e80b.worker.js:1:35651)
    at a (babel-transpiler.2641e80b.worker.js:1:35862)
    at babel-transpiler.2641e80b.worker.js:1:35921
    at new Promise (<anonymous>)
    at babel-transpiler.2641e80b.worker.js:1:35802
    at Qe (babel-transpiler.2641e80b.worker.js:1:1089330)
    at babel-transpiler.2641e80b.worker.js:1:1094183
    at Generator.next (<anonymous>)
    at n (babel-transpiler.2641e80b.worker.js:1:35651)
    at a (babel-transpiler.2641e80b.worker.js:1:35862)
(anonymous) @ babel-transpiler.2641e80b.worker.js:1Understand this errorAI
vendors~sandbox-startup.ca8a95b40.chunk.js:1 Error in sandbox:
vendors~sandbox-startup.ca8a95b40.chunk.js:1 TypeError: Cannot assign to read only property 'message' of object 'SyntaxError: /App.tsx: Unexpected token (2:0)

  1 | import { useState,
> 2 |
    | ^'
    at sandbox.bc50c697e.js:1:251961
    at ze.handleCallResponse (sandbox.bc50c697e.js:1:252080)
    at ze.handleMessage (sandbox.bc50c697e.js:1:250575)
```


### Screenshots

<img width="1726" alt="image" src="https://github.com/user-attachments/assets/6643489d-21d5-43c3-bddd-6065db06e940" />

<img width="1159" alt="image" src="https://github.com/user-attachments/assets/7a454f92-4b9c-460a-a475-aa8502da4d9f" />


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/api/app/clients/BaseClient.js
+++ b/api/app/clients/BaseClient.js
@@ -931,6 +931,24 @@ class BaseClient {
             continue;
           }
 
+          if (item.type === 'tool_call' && item.tool_call != null) {
+            const toolName = item.tool_call?.name || '';
+            if (toolName != null && toolName && typeof toolName === 'string') {
+              numTokens += this.getTokenCount(toolName);
+            }
+
+            const args = item.tool_call?.args || '';
+            if (args != null && args && typeof args === 'string') {
+              numTokens += this.getTokenCount(args);
+            }
+
+            const output = item.tool_call?.output || '';
+            if (output != null && output && typeof output === 'string') {
+              numTokens += this.getTokenCount(output);
+            }
+            continue;
+          }
+
           const nestedValue = item[item.type];
 
           if (!nestedValue) {

--- a/api/server/utils/import/importers.js
+++ b/api/server/utils/import/importers.js
@@ -113,7 +113,7 @@ async function importLibreChatConvo(
        */
       const traverseMessages = async (messages, parentMessageId = null) => {
         for (const message of messages) {
-          if (!message.text) {
+          if (!message.text && !message.content) {
             continue;
           }
 
@@ -121,6 +121,7 @@ async function importLibreChatConvo(
           if (message.sender?.toLowerCase() === 'user' || message.isCreatedByUser) {
             savedMessage = await importBatchBuilder.saveMessage({
               text: message.text,
+              content: message.content,
               sender: 'user',
               isCreatedByUser: true,
               parentMessageId: parentMessageId,
@@ -128,6 +129,7 @@ async function importLibreChatConvo(
           } else {
             savedMessage = await importBatchBuilder.saveMessage({
               text: message.text,
+              content: message.content,
               sender: message.sender,
               isCreatedByUser: false,
               model: options.model,

--- a/client/src/components/Chat/Input/Files/FilePreview.tsx
+++ b/client/src/components/Chat/Input/Files/FilePreview.tsx
@@ -21,7 +21,11 @@ const FilePreview = ({
 }) => {
   const radius = 55; // Radius of the SVG circle
   const circumference = 2 * Math.PI * radius;
-  const progress = useProgress(file?.['progress'] ?? 1, 0.001, (file as ExtendedFile).size ?? 1);
+  const progress = useProgress(
+    file?.['progress'] ?? 1,
+    0.001,
+    (file as ExtendedFile | undefined)?.size ?? 1,
+  );
 
   // Calculate the offset based on the loading progress
   const offset = circumference - progress * circumference;

